types/pico8parse/index.d.ts

```diff
@@ -0,0 +1,129 @@
+     * and how Lua escape sequences in JavaScript strings should be interpreted.
+     * See the Encoding modes section https://github.com/fstirlitz/luaparse#encoding-modes for more information.
+     */
+    encodingMode: "pseudo-latin1" | "x-user-defined" | "none";
+    /**
+     * This option should be reserved for testing but may be use if needed;
+     * it overrides the `strictP8FileFormat` feature, making it possible to parse
+     * snippets lacking the proper header and sections
+     */
+    ignoreStrictP8FileFormat: boolean;
+}
+
+export interface Parser {
+    version: string;
+    /**
+     * As this parser is a bit different from Lua's own, the error messages
+     * will be different in some situations.
+     *
+     * This maps from an error key to a user-friendly string describing the error.
+     * Each string may contain "%n" sequences to be interpolated with error-dependent
+     * contextual information (n is an integer, counting from 1).
+     */
+    errors: Record<string, string>;
+
+    /**
+     * The main parser.
+     * @example
+     * var parser = require('pico8parse');
+     * parser.parse('i = 0');
+     */
+    parse(code: string, options: Partial<Options> & { wait: true }): Parser;
+    parse(code: string, options?: Partial<Options>): Chunk;
+    parse(options?: Partial<Options>): Parser;
+    /** Write to the source code buffer without beginning the parse. */
+    write(segment: string): Parser;
+    /** Send an EOF and begin parsing. */
+    end(segment: string): Chunk;
+    /**
+     * The lexer, or the tokenizer reads the input string character by character
+     * and derives a token left-right. To be as efficient as possible the lexer
+     * prioritizes the common cases such as identifiers. It also works with
+     * character codes instead of characters as string comparisons was the
+     * biggest bottleneck of the parser.
+     *
+     * If `options.comments` is enabled, all comments encountered will be stored
+     * in an array which later will be appended to the chunk object. If disabled,
+     * they will simply be disregarded.
+     *
+     * When the lexer has derived a valid token, it will be returned as an object
+     * containing its value and as well as its position in the input string (this
+     * is always enabled to provide proper debug messages).
+     *
+     * `lex()` starts lexing and returns the following token in the stream.
+     */
+    lex(): Token;
+
+    defaultOptions: Omit<Options, 'extendedIdentifiers'>;
+    versionFeatures: Record<LuaVersion, boolean>;
+}
+
+/**
+ * This is temporary and may be changed
+ *
+ * prefer instanceof checking against this rather than the engine built-in SyntaxError
+ * (see https://github.com/fstirlitz/luaparse/releases/tag/v0.3.1 and #67)
+ */
+export class SyntaxError extends Error {}
+
+declare const parser: Parser;
+export default parser;
```

Comment:
This would be written better as a UMD definition, since [it’s published that way](https://unpkg.com/browse/pico8parse@0.3.1001/luaparse.js).